lib/pacemaker/pcmk_sched_location.c

```diff
@@ -227,29 +216,13 @@ generate_location_rule(pcmk_resource_t *rsc, xmlNode *rule_xml,
         if (rc == pcmk_rc_ok) {
             pcmk_node_t *local = pe_find_node_id(nodes, node->details->id);
 
-            if ((local == NULL) && (combine == pcmk__combine_and)) {
-                continue;
-
-            } else if (local == NULL) {
+            if (local == NULL) {
```

Comment:
I think we can make this even simpler. We're iterating over `rsc->cluster->nodes` and adding them to our `nodes` list one-by-one if appropriate.

Unless there are duplicate nodes within the scheduler's node set, `local` will always be `NULL` here. How would it have made it into `nodes` before this point?

(Also if that weren't the case, then our finding about overwriting the score would not have held up.)

Reply:
yeah it can actually simplify a whole lot more